vizro-core/src/vizro/models/_components/form/switch.py

```diff
@@ -0,0 +1,109 @@
+from typing import Annotated, Any, Literal, Optional
+
+import dash_bootstrap_components as dbc
+from dash import html
+from pydantic import AfterValidator, BeforeValidator, Field, PrivateAttr
+from pydantic.functional_serializers import PlainSerializer
+from pydantic.json_schema import SkipJsonSchema
+
+from vizro.models import Tooltip, VizroBaseModel
+from vizro.models._action._actions_chain import _action_validator_factory
+from vizro.models._components.form._form_utils import validate_value
+from vizro.models._models_utils import warn_description_without_title
+from vizro.models._tooltip import coerce_str_to_tooltip
+from vizro.models.types import ActionType, StrictBool, _IdProperty
+
+
+class Switch(VizroBaseModel):
+    """Boolean single-option selector `Switch`.
+
+    Can be provided to [`Filter`][vizro.models.Filter] or
+    [`Parameter`][vizro.models.Parameter].
+
+    Args:
+        type (Literal["toggle_switch"]): Defaults to `"toggle_switch"`.
+        value (Optional[StrictBool]): See [`StrictBool`][vizro.models.types.StrictBool]. Defaults to `False`.
+        title (str): Title to be displayed. Defaults to `""`.
+        description (Optional[Tooltip]): Optional markdown string that adds an icon next to the title.
+            Hovering over the icon shows a tooltip with the provided description. Defaults to `None`.
+        actions (list[ActionType]): See [`ActionType`][vizro.models.types.ActionType]. Defaults to `[]`.
+        extra (Optional[dict[str, Any]]): Extra keyword arguments that are passed to `dbc.RadioItems` and overwrite any
+            defaults chosen by the Vizro team. This may have unexpected behavior.
+            Visit the [dbc documentation](https://dash-bootstrap-components.opensource.faculty.ai/docs/components/input/)
```

Comment:
Can you check if the link is corect? It doesn't seem to work for me 🤔 
```suggestion
            Visit the [dbc documentation](https://dash-bootstrap-components.opensource.faculty.ai/docs/components/input/)
```